packages/ckeditor5-link/src/autolink.js

```diff
@@ -0,0 +1,207 @@
+/**
+ * @license Copyright (c) 2003-2020, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module link/autolink
+ */
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+import TextWatcher from '@ckeditor/ckeditor5-typing/src/textwatcher';
+import getLastTextLine from '@ckeditor/ckeditor5-typing/src/utils/getlasttextline';
+
+const MIN_LINK_LENGTH_WITH_SPACE_AT_END = 4; // Ie: "t.co " (length 5).
+
+const URL_REG_EXP = new RegExp(
+	// Group 1: Line start or after a space.
+	'(^|\\s)' + // Match .
+	// Group 2: Full detected URL.
+	'(' +
+		// Group 3 + 4: Protocol + domain.
+		'(([a-z]{3,9}:(?:\\/\\/)?)(?:[\\w]+)?[a-z0-9.-]+|(?:www\\.|[\\w]+)[a-z0-9.-]+)' +
+		// Group 5: Optional path + query string + location.
+		'((?:\\/[+~%/.\\w\\-_]*)?\\??(?:[-+=&;%@.\\w_]*)#?(?:[.!/\\\\\\w]*))?' +
+	')$', 'i' );
+
+const URL_GROUP_IN_MATCH = 2;
+
+// Simplified email test - should be run over previously found URL.
+const EMAIL_REG_EXP = /^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,}$/i;
+
+/**
+ * The auto link plugin.
+ *
+ * @extends module:core/plugin~Plugin
+ */
+export default class AutoLink extends Plugin {
```

Comment:
The plugin uses a few things that are provided by other plugins and as I can see, the dependencies are not specified:

* `editor.commands.get( 'enter' )` - what if missing? I know, little chance for such case but still...
* `const shiftEnterCommand = editor.commands.get( 'shiftEnter' );` - the same as above,
* `writer.setAttribute( 'linkHref', linkHrefValue, range );` - missing schema definition or `LinkEditing` which defines it.

Reply:
I've added a tests for `Enter` and `ShiftEnter` - I've forgot about that.

The feature is build similarly to `AutoMediaEmbed` - it assumes `MediaEmbedEditing` to be present. It uses `linkHref` as `editor.execute( 'link' )` does not accept range :/

Reply:
Added here: #7490.